Autopilot/PathManager/Src/cruisingState.cpp

```diff
@@ -0,0 +1,349 @@
+        
+        // Updates the in hold flag if execution was successful
+        if (inHold && pathFollowingStatus == WAYPOINT_SUCCESS) {
+            inHold = false;
+        } else if (!inHold && pathFollowingStatus == WAYPOINT_SUCCESS) {
+            inHold = true;
+        }
+    } else if (telemetryData->waypointNextDirectionsCommand == 2) { // Heading home
+
+        // cout << "Flight Path: going home" << endl;
+
+        // Sees if the plane is currently going home. If it is, then we will cancel. Else, start going home
+        if (goingHome) {
+            goingHome = false;
+        } else {
+            goingHome = true;
+        }
+
+        goingHomeStatus = cruisingStateManager.head_home(goingHome);
+
+        if (goingHomeStatus == HOME_UNDEFINED_PARAMETER && goingHome) { // If setting home mode fails, reverse change
+            goingHome = false;
+            pathFollowingStatus = UNDEFINED_PARAMETER;
+        } else if (goingHomeStatus == HOME_TRUE || goingHomeStatus == HOME_FALSE) {
+            pathFollowingStatus = WAYPOINT_SUCCESS;
+        }
+    } else { // Incorrect commands from telemetry
+
+        // cout << "Flight Path: incorrect command" << endl;
+
+        telemetryData->waypointNextDirectionsCommand = 0; // Set important values to their defaults. This will ensure that if the telemetry struct is not change, our plane will behave as expected
+
+        return 2;
+    }
+
+    telemetryData->waypointNextDirectionsCommand = 0;  // Set important values to their defaults. This will ensure that if the telemetry struct is not change, our plane will behave as expected
+
+    // Return appropriate error code
+    if (pathFollowingStatus == WAYPOINT_SUCCESS) {
+        return 0;
+    } else if (goingHomeStatus == HOME_UNDEFINED_PARAMETER) {
+        return 3;
+    } else {
+        return 1;
+    }
+}
+
+void setReturnValues(_CruisingState_Telemetry_Return * _returnToGround, WaypointManager& cruisingStateManager, int editErrorCode, int pathErrorCode) {
+    _returnToGround->currentWaypointId = cruisingStateManager.get_id_of_current_index();
+    _returnToGround->currentWaypointIndex = cruisingStateManager.get_current_index();
+    _returnToGround->homeBaseInitialized = cruisingStateManager.is_home_base_initialized();
+
+    _returnToGround->editingFlightPathErrorCode = editErrorCode;
+    _returnToGround->pathFollowingErrorCode = pathErrorCode;
+}
+
+void appendNewElement(int * idArray, int newId) {
+    int counter = 0;
+    bool appended = false;
+
+    // Goes through array until empty index is found or we reach end of array
+    while (!appended && counter < PATH_BUFFER_SIZE) {
+        if (idArray[counter] == 0) { // If an element has a value of 0, it is empty
+            idArray[counter] = newId;
+            appended = true;
+        }
+
```

Comment:
remove spacing

Reply:
Ngl, space is usually a good thing to have provided it doesn't impact readability.

During compilation all white space is removed so it doesn't impact the code's performance anyway.